### ISSUE TITLE
Error with createUploadDestinationForResource

### ISSUE THREAD

**Issue by OWLMicha** [jlevers/selling-partner-api#241]
Hello,

I'm brand new here and I'm not a professional programmer either, but I've already had good success with the SDK, thank you for the good work.

However, I now have a problem that I can't solve and I think it's a bug.

### **createUploadDestinationForResource**

This is the error code I'm getting:

_Exception when calling UploadsApi->createUploadDestinationForResource: [403] { "errors": [ { "message": "The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation The Canonical String for this request should have been 'POST /uploads/2020-11-01/uploadDestinations/contentDocuments_

It is :
` $result = $apiInstance->createUploadDestinationForResource($marketplace_ids, $content_md5, $resource, $content_type);`

Otherwise everything works great and I only have access via the API with this request, are there problems, so I think it might be a bug in the code?

I would be very happy about support.

Best regards

### Addendum:

Additional information that may be important:

**I want to upload images to A+ Content.**

Here the resource is apparently also a different one:

[LINK A+ Case Guide](https://developer-docs.amazon.com/sp-api/docs/a-content-api-use-case-guide#Task-2-Create-upload-destinations-for-images)

_The URL of the resource for the upload destination that you are creating. For example, to create an upload destination for a content document, the {resource} would be /contentDocuments and the path would be /uploads/v1/uploadDestinations/aplus/2020-11-01/contentDocuments_

Type: string
![image](https://user-images.githubusercontent.com/103287033/163193352-be52bd75-2baa-4446-98ef-6964ea9d42c2.png)

I hope that someone can help me.

Thanks and best regards

**Comment by OWLMicha** [comment]
Hello,
I think I was able to fix the problem first.

So I get the desired result, because I'm not a professional, I think that you can definitely solve it differently and better in the next version.

in the UploadsAPI.php file



**OLD (  '{' . 'resource' . '}', ohne /   ) :**
```
  if ($resource !== null) {
             $resourcePath = str_replace(
                 '{' . 'resource' . '}',
                 ObjectSerializer::toPathValue($resource, true),
                 $resourcePath
             );
         }
      
```

**New  (   '/{' . 'resource' . '}', mit /  )  :**

```
  if ($resource !== null) {
             $resourcePath = str_replace(
                 '/{' . 'resource' . '}',
                 ObjectSerializer::toPathValue($resource, true),
                 $resourcePath
             );
         }
```

maybe it will help someone and it can be adjusted in the next version.
Best regards Micha

**Comment by jlevers** [comment]
Interesting, this same call had another problem (#176), but I think we got it working. What happens if you submit the request using the original code, but without the leading slash (e.g., `contentDocuments` instead of `/contentDocuments`)?

**Comment by OWLMicha** [comment]
Unfortunately, I can no longer reproduce it 100% because I have tried a lot and also switched from an older to a new version.

However, I can't recreate it right now and I don't know if the error wasn't somewhere else with the new version.

**Here is the correct value with which the query works:**
`$resource = 'aplus/2020-11-01/contentDocuments'; `

**Here the description from amazon is also somewhat your guide. (See image.)**
![image](https://user-images.githubusercontent.com/103287033/163376559-55ecc681-b615-4faa-900f-1a29ab4267c2.png)



**These were my error messages:**
`Exception when calling UploadsApi->createUploadDestinationForResource: [403] { "errors": [ { "message": "The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details. The Canonical String for this request should have been 'POST /uploads/2020-11-01/uploadDestinations/aplus/2020-11-01/contentDocuments contentMD5=2f9b80fcf031f80c481b66f46d729f7f&contentType=image%2Fjpeg&marketp`


`Exception when calling UploadsApi->createUploadDestinationForResource: [403] { "errors": [ { "message": "Access to requested resource is denied.", "code": "Unauthorized", "details": "" } ] }`
